### PR TITLE
Submit and assert a single parameter value

### PR DESCRIPTION
**Pull request by GernotMaier**
This PR is a follow up PR of #490: submit and validate a single parameter (#490 adds code to validate tabled data).

Input data can be given in a simple yml file, e.g.:
```
---
name: ref_long
value: [-17.8920302]
units: [deg]
...
```
or as the testing file [tests/resources/reference_position_mercator.yml](tests/resources/reference_position_mercator.yml). Schema are given in the usual format (see e.g., [tests/resources/reference_position_mercator.schema.yml](tests/resources/reference_position_mercator.schema.yml).

Note the simplification of the approach in the code by converting input data to an astropy table (in this case with one row and one column). So all validation routines are identical to #490, most changes introduces here are about reading the yaml file and converting the data to tabled data.

The input data is compared to the schema, validated for units and range, and then written as a astropy table to disk (by default: ecsv). The latter is something which is probably preliminary, as we want to write model parameters into the database.

Note the addition of code to find the schema files:
- `schema` configuration parameter is a file: this is the schema file
- `schema` configuration parameter is a directory: read parameter name from data file and search for `schema/<parameter name>.schema.yml`. This will simplify workflows and points towards the intended file naming for schema files.

### CODE REVIEW

File: simtools/data_model/validate_data.py
Comment (orelgueta) [inline simtools/data_model/validate_data.py:110]: "tabled data" reads a bit strange. Should it be "tabulated data" or just "table data"? Google doesn't give many results for "tabled data" so I think it's not just me.

File: simtools/data_model/validate_data.py
Comment (orelgueta) [inline simtools/data_model/validate_data.py:399]: `f-string`

File: simtools/data_model/validate_data.py
Comment (orelgueta) [inline simtools/data_model/validate_data.py:420]: `f-string`

File: simtools/data_model/validate_data.py
Comment (orelgueta) [inline simtools/data_model/validate_data.py:546]: `f-string`

Comment (GernotMaier) [reply simtools/data_model/validate_data.py:546]: Fixed.

File: simtools/data_model/validate_data.py
Comment (GernotMaier) [reply simtools/data_model/validate_data.py:490]: Done.

File: simtools/data_model/validate_data.py
Comment (GernotMaier) [reply simtools/data_model/validate_data.py:420]: Fixed.

File: simtools/data_model/validate_data.py
Comment (GernotMaier) [reply simtools/data_model/validate_data.py:399]: Fixed.

File: simtools/data_model/validate_data.py
Comment (GernotMaier) [reply simtools/data_model/validate_data.py:119]: Fixed.

File: simtools/data_model/validate_data.py
Comment (GernotMaier) [reply simtools/data_model/validate_data.py:103]: True, missed this. Added it.

File: simtools/data_model/validate_data.py
Comment (GernotMaier) [reply simtools/data_model/validate_data.py:79]: Fixed.

**Comment by GernotMaier** [comment]
This is ready for the second review. Important changes:

- general.py has a new function collect_dict_from_file which allows to read a dict from a file or from a path + filename. This removes duplicated code in validate_data and metadata_collector
- required some adjustments to tests
- adjustments to the new simplified schema for parameter schema.yml files (also updated the example in tests/resources)
- download jsonschema.yml now from github webpage (before: was in test DB; which required to make sure that this DB entry is always up-to-date). Note! for now I am reading this file from a branch in the workflows repo until [this PR](https://github.com/gammasim/workflows/pull/11) is merged. 
- no changes to filenames. It continues to be `.schema.yml`. This requires more discussions and I don't agree that we should do it (e.g., there is `.simtel.zst`)

Even this PR can be reviewed, I will not merge not after approval but wait for the PR in workflows to fix the url in the integration tests.

File: simtools/data_model/validate_data.py
Comment (orelgueta) [inline simtools/data_model/validate_data.py:473]: You changed the docstring but decided against changing the code afterwards =)

File: tests/unit_tests/data_model/test_metadata_collector.py
Comment (orelgueta) [inline tests/unit_tests/data_model/test_metadata_collector.py:239]: Are you sure you need to use `OrderedDict` here instead of just `dict`?

**Review by orelgueta — COMMENTED** [review]
Very minor comments left. I could already approve technically but I would like you to check the changes I made as well.

Regarding `.schema.yml` vs. `_schema.tml`, I like the fact that we both used the `.simtel.zst` example to make our opposing points =)

**Comment by GernotMaier** [comment]
Changes have been submitted. Let me know if there is anything else.

**Review by orelgueta — APPROVED** [review]

**Pull request merged**
